plugins/woocommerce-blocks/assets/js/blocks/product-filters/inner-blocks/attribute-filter/edit.tsx

```diff
@@ -69,144 +76,181 @@ const Edit = ( props: EditProps ) => {
 			isEditor: true,
 		} );
 
-	const isLoading = isTermsLoading || isCountsLoading;
-
 	useEffect( () => {
+		if ( isTermsLoading || isFilterCountsLoading ) return;
+
 		const termIdHasProducts =
 			objectHasProp( filteredCounts, 'attribute_counts' ) &&
 			isAttributeCounts( filteredCounts.attribute_counts )
 				? filteredCounts.attribute_counts.map( ( term ) => term.term )
 				: [];
 
-		if ( termIdHasProducts.length === 0 && hideEmpty )
-			return setAttributeOptions( [] );
-
-		setAttributeOptions(
-			attributeTerms
-				.filter( ( term ) => {
-					if ( hideEmpty )
-						return termIdHasProducts.includes( term.id );
-					return true;
-				} )
-				.sort( ( a, b ) => {
-					switch ( sortOrder ) {
-						case 'name-asc':
-							return a.name > b.name ? 1 : -1;
-						case 'name-desc':
-							return a.name < b.name ? 1 : -1;
-						case 'count-asc':
-							return a.count > b.count ? 1 : -1;
-						case 'count-desc':
-						default:
-							return a.count < b.count ? 1 : -1;
-					}
-				} )
-		);
-	}, [ attributeTerms, filteredCounts, sortOrder, hideEmpty ] );
+		if ( termIdHasProducts.length === 0 && hideEmpty ) {
+			setAttributeOptions( [] );
+		} else {
+			setAttributeOptions(
+				attributeTerms
+					.filter( ( term ) => {
+						if ( hideEmpty )
+							return termIdHasProducts.includes( term.id );
+						return true;
+					} )
+					.sort( ( a, b ) => {
+						switch ( sortOrder ) {
+							case 'name-asc':
+								return a.name > b.name ? 1 : -1;
+							case 'name-desc':
+								return a.name < b.name ? 1 : -1;
+							case 'count-asc':
+								return a.count > b.count ? 1 : -1;
+							case 'count-desc':
+							default:
+								return a.count < b.count ? 1 : -1;
+						}
+					} )
+					.map( ( term, index ) => ( {
+						label: showCounts
+							? `${ term.name } (${ term.count })`
+							: term.name,
+						value: term.id.toString(),
+						selected: index === 1,
+						rawData: term,
+					} ) )
+			);
+		}
 
-	const Wrapper = ( { children }: { children: React.ReactNode } ) => (
-		<div { ...useBlockProps() }>
-			<Inspector { ...props } />
-			{ children }
-		</div>
+		setIsOptionsLoading( false );
+	}, [
+		showCounts,
+		attributeTerms,
+		filteredCounts,
+		sortOrder,
+		hideEmpty,
+		isTermsLoading,
+		isFilterCountsLoading,
+	] );
+
+	const { children, ...innerBlocksProps } = useInnerBlocksProps(
+		useBlockProps(),
+		{
+			allowedBlocks: getAllowedBlocks( DISALLOWED_BLOCKS ),
```

Comment:
I think this is a bit confusing you're trying to get allowed blocks with a constant that says `DISALLOWED_BLOCKS`